files/check_license.sh

```diff
@@ -0,0 +1,51 @@
+#!/usr/local/bin/bash
+# File to check
+declare -r LICENSE_FILE="/usr/local/opnsense/version/core.license"
+
+# Function to output error messages and exit
+function error_exit {
+    local message="$1"
+    echo "$message"
+    exit 3
+}
+
+# Ensure the file exists
+if [[ ! -f "$LICENSE_FILE" ]]; then
+    # no license file means community edition
+    exit
+fi
+
+# Extract "valid_to" from the license file
+VALID_TO=$(awk -F'"' '/"valid_to"/ {print $4}' "$LICENSE_FILE")
```

Comment:
```suggestion
VALID_TO=$(jq .valid_to "$LICENSE_FILE")
```

Reply:
I think using `jq` instead of `awk` makes this more robust;
in case some opnsense developer decides to change the syntax by adding some newlines etc. the file will not change in terms of json syntax, but might give a different output using awk ...

Reply:
so @mnaier : great improvements, thank you, but especially the openvpn checks needs some changes in order to be more generic

Reply:
@mnaier you are right, thank you for your attention
I thought `jq` is installed by default, but it seems I was wrong, just  `json_pp` is there by default.
:+1: